### PR TITLE
CFY-7156 Delete init/install scripts after they've been used

### PR DESCRIPTION
**Pull request by tehasdf**

### CODE REVIEW

File: cloudify_agent/installer/script.py [hunk 1]
```diff
@@ -203,3 +207,27 @@ def install_script_path(cloudify_agent):
...
         os.remove(script_path)
+
+
+def _cleanup_after_installation(path):
+    """Mark path to be deleted after agent installation.
+
+    This simply adds the path to cloudify_agent inside runtime properties,
+    so that it can be removed later.
+    """
+    cloudify_agent = ctx.instance.runtime_properties.get('cloudify_agent', {})
```
Comment (mcouthon) [inline cloudify_agent/installer/script.py]: This seems unnecessary - you could just pass `self.cloudify_agent` from the above class

File: cloudify_agent/installer/script.py [hunk 2]
```diff
@@ -203,3 +207,27 @@ def install_script_path(cloudify_agent):
...
+    cleanup = cloudify_agent.get(LOCAL_CLEANUP_PATHS_KEY, [])
+    cleanup.append(path)
+    cloudify_agent[LOCAL_CLEANUP_PATHS_KEY] = cleanup
+    ctx.instance.runtime_properties['cloudify_agent'] = cloudify_agent
+
+
+def cleanup_scripts():
+    """Remove the files that were scheduled for deletion."""
+    cloudify_agent = ctx.instance.runtime_properties.get('cloudify_agent', {})
+    paths = cloudify_agent.pop(LOCAL_CLEANUP_PATHS_KEY, [])
```
Comment (mcouthon) [inline cloudify_agent/installer/script.py]: Does this pop gets registered by the ctx and the runtime props?

File: cloudify_agent/installer/script.py [hunk 3]
```diff
@@ -203,3 +207,27 @@ def install_script_path(cloudify_agent):
...
+    """Mark path to be deleted after agent installation.
+
+    This simply adds the path to cloudify_agent inside runtime properties,
+    so that it can be removed later.
+    """
+    cloudify_agent = ctx.instance.runtime_properties.get('cloudify_agent', {})
+    cleanup = cloudify_agent.get(LOCAL_CLEANUP_PATHS_KEY, [])
+    cleanup.append(path)
+    cloudify_agent[LOCAL_CLEANUP_PATHS_KEY] = cleanup
+    ctx.instance.runtime_properties['cloudify_agent'] = cloudify_agent
```
Comment (mcouthon) [inline cloudify_agent/installer/script.py]: Here you can use the `update_agent_runtime_properties` func from `agent_config.py`

File: cloudify_agent/installer/script.py (hunk 2 above)
Comment (tehasdf) [reply cloudify_agent/installer/script.py]: it doesn't, added a `update` there

**Pull request merged**